### PR TITLE
Implement SdJwtVP.of(String) with enhanced error handling

### LINKED ISSUE #33020: Incorrect Disclosure Handling in SdJwtVP.of(String) Method
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

There is parsing error in the "SdJwtVP.java"  file when the length of 'disclosureStart' equals 'disclosureEnd'. In this case, the value should be empty, and the assignment should only occur when 'End > Start'. 

https://github.com/keycloak/keycloak/blob/5fe916861df60a8791b14d84474eaa517ca97c81/core/src/main/java/org/keycloak/sdjwt/vp/SdJwtVP.java#L104-L110

### Version

25.0.5

### Regression

- [ ] The issue is a regression

### Expected behavior

When "disclosureStart"  equals "disclosureEnd", the value should be empty. The assignment should occur when End > Start.

### Actual behavior

An IllegalArgumentException" is thrown with the misleading error message " ("SD-JWT is malformed, expected to end with " + SdJwt.DELIMITER) ".

### How to Reproduce?

Create an SD-JWT wwithout the delimiter at the end (e.g., "issuer-signed-jwt" instead of "issuer-signed-jwt).

Attempt to parse the SD-JWT using the `SdJwtVp.of()` method.

### Anything else?

I'll send a PR to address the issue.

### PR DESCRIPTION
**Pull request by Ogenbertrand**
This PR implements the SdJwtVP.of(String) method with improved error handling. It adds validation to handle missing disclosures, and invalid disclosure data during the disclosure processing flow.

Changes Implemented
Added a check for missing disclosures, which throws an IllegalArgumentException if none are found.
Improved error handling for invalid or malformed disclosure data by throwing appropriate exceptions.

Testing
Unit tests have been added to cover the following cases:
Missing disclosures
Invalid/malformed disclosure data

Issue Reference
Closes #33020

### CODE REVIEW

**Review by francis-pouatcha — APPROVED** [review]

**Review by IngridPuppet — APPROVED** [review]

**Comment by francis-pouatcha** [comment]
@ahus1 is there anything missing to merge this pull request?

**Review by ahus1 — COMMENTED** [review]
Thank you for the PR. See below for a something that I think is a mismatch of the check and the error message. 

Not sure if you want to change the check, or the error message. As I'm not into SDJ-JWT , I'm marking it as a comment and leave it to you to find the answer.

**Comment by ahus1** [comment]
@francis-pouatcha / @IngridPuppet - please review the latest changes if they look good to you. Thanks!

File: core/src/main/java/org/keycloak/sdjwt/vp/SdJwtVP.java
Comment (IngridPuppet) [inline core/src/main/java/org/keycloak/sdjwt/vp/SdJwtVP.java:112]: Better with spaces around the equal sign.
```suggestion
        String disclosuresString = "";
```

File: core/src/test/java/org/keycloak/sdjwt/sdjwtvp/SdJwtVPTest.java
Comment (IngridPuppet) [inline core/src/test/java/org/keycloak/sdjwt/sdjwtvp/SdJwtVPTest.java:219]: This test will likely fail as is due to the asserted error message not aligning with the changes. Additionally, the test vector should have a different comment, since as highlighted by  @ahus1, the problem is not that the SD-JWT is not ending with ~.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]
Thank you @ahus1 for the extra attention, and thank you @Ogenbertrand for addressing the comments. I have some minor additional concerns.

File: core/src/test/java/org/keycloak/sdjwt/sdjwtvp/SdJwtVPTest.java
Comment (Ogenbertrand) [reply core/src/test/java/org/keycloak/sdjwt/sdjwtvp/SdJwtVPTest.java:219]: @IngridPuppet Thank you for pointing out the issue with the test. I'll update the test to reflect the changes made to the error message. I'll make sure to address this in the next update.

**Comment by ahus1** [comment]
@Ogenbertrand - please avoid rebasing this PR or merging the main branch into this PR as it will reset the CI jobs every time. This is a waste of CI resources, and it will also reset any previous review. 

Don't rebase a PR unless there are conflicts or a maintainer asks you to rebase the PR.

**Comment by Ogenbertrand** [comment]
> @Ogenbertrand - please avoid rebasing this PR or merging the main branch into this PR as it will reset the CI jobs every time. This is a waste of CI resources, and it will also reset any previous review.
> 
> Don't rebase a PR unless there are conflicts or a maintainer asks you to rebase the PR.

Okay @ahus1, Understood, i was actually trying to fix the error with the previous run.

**Review by IngridPuppet — APPROVED** [review]

**Review by francis-pouatcha — APPROVED** [review]

**Review by tnorimat — APPROVED** [review]

**Review by ahus1 — APPROVED** [review]
Approving based on previews reviews. Thanks!

**Pull request merged**
